### PR TITLE
Table relationships

### PR DESCRIPTION
**Pull request by LucyMHall**
This branch has the following functionality:

- It links the post and user databases (has_many, belongs_to relationships) so that when a post is created it is associated with the current user
- It reflects the database relationship in the routes by having posts listed as a nested resource in the config routes file
- It alters the index page for posts so that it is tied to a particular user
- It adds a button to the index page so a user can navigate to their posts when they log in
- It is fully feature tested

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -40,6 +40,7 @@ group :development, :test do
   gem 'selenium-webdriver'
   gem 'simplecov', require: false
   gem 'rspec-rails', '~> 3.5'
+  gem 'database_cleaner'
```
Comment (rachjgriff) [inline Gemfile]: can't wait to hear about this!

File: app/views/posts/new.html.erb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<%= form_for @post do |form| %>
+<%= form_with(model: [current_user.posts.build], local: true) do |form| %>
```
Comment (rachjgriff) [inline app/views/posts/new.html.erb]: what does the local bit mean??

File: config/routes.rb [hunk 3]
```diff
@@ -1,8 +1,9 @@
 Rails.application.routes.draw do
   devise_for :users
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
-
-  resources :posts
+  resources :users do
```
Comment (rachjgriff) [inline config/routes.rb]: nice nesting!

File: db/schema.rb [hunk 4]
```diff
@@ -19,6 +19,8 @@
     t.string "message"
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
+    t.bigint "user_id"
```
Comment (rachjgriff) [inline db/schema.rb]: what is t.bigint???

File: spec/support/database_cleaner.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
+RSpec.configure do |config|
```
Comment (rachjgriff) [inline spec/support/database_cleaner.rb]: did this come with the gem or did you have to set this up?

File: db/schema.rb (hunk 4 above)
Comment (Danielandro) [reply db/schema.rb]: It is a Postgres data type - allows for really big integers (assuming we gain millions of Acebook users!)
It allows numbers btwn **-9,223,372,036,854,775,808** &  **+9,223,372,036,854,775,807**

**Pull request merged**